src/Common/DNSResolver.cpp

```diff
@@ -198,21 +194,90 @@ struct DNSResolver::Impl
     std::atomic<bool> disable_cache{false};
 };
 
+struct DNSResolver::AddressFilter
+{
+    struct DNSFilterSettings
+    {
+        std::atomic<bool> dns_allow_resolve_names_to_ipv4{true};
```

Comment:
Consider using [`MultiVersion<T>`](https://github.com/aalexfvk/ClickHouse/blob/74b870710f51a43c5cca4d42119074250433ef3e/src/Common/MultiVersion.h#L29) to update all settings as a whole